src/Model/CommonTypes.hs

```diff
@@ -141,12 +194,20 @@ type Map = [MapRow]
 
 type MapRow = [Tile]
 
-data TileType = Solid | Transparent
+data Tile = Solid Appearance | Transparent Appearance
   deriving (Show)
 
-data Tile = Tile
-  { tileType :: TileType
-  , tileObject :: Object
-  } deriving (Show)
+instance Eq Tile where
+  (==) (Solid _) (Transparent _) = False
+  (==) (Transparent _) (Solid _) = False
+  (==) (Solid a1) (Solid a2) = appearanceBox a1 == appearanceBox a2
+  (==) (Transparent a1) (Transparent a2) = appearanceBox a1 == appearanceBox a2
+
+isSolid (Solid _) = True
+isSolid (Transparent _) = False
```

Comment:
Change `(Tranparent _)` to just `_` for future extensibility.

Reply:
done